agent/consul/internal_endpoint.go

```diff
@@ -178,11 +178,19 @@ func (m *Internal) KeyringOperation(
 		}
 	}
 
-	// Only perform WAN keyring querying and RPC forwarding once
-	if !args.Forwarded && m.srv.serfWAN != nil {
-		args.Forwarded = true
-		m.executeKeyringOp(args, reply, true)
-		return m.srv.globalRPC("Internal.KeyringOperation", args, reply)
+	// Validate use of local-only
+	if args.LocalOnly && args.Operation != structs.KeyringList {
+		// Error aggressively to be clear about LocalOnly behavior
+		return fmt.Errorf("argument error: LocalOnly can only be used for List operations")
+	}
+
+	if !args.LocalOnly { // LocalOnly should always be false for non-GET requests
```

Comment:
I don't have super strong opinions here but stylistically I think this comment would be better on the line preceding the `if`. At least in much of the Consul codebase we don't typically have comments on the end of a non-comment line (except for documenting struct fields).

Reply:
got it! will do